app/assets/stylesheets/generic.sass.scss

```diff
@@ -56,30 +56,23 @@ dl.content {
     display: none;
   }
 
-  #container {
-    width: 950px;
-    margin: 0 auto;
-    display: grid;
-    grid-gap: 1rem;
-    color: $link-hover-colour;
-  }
-
   body {
     display: flex;
     flex-direction: column;
   }
 
-  #container {
-    grid-template-columns: 196px 650px;
-    background: transparent url("page-bg.png") repeat-x;
+  .main-outer-container {
     flex: 1 0 auto;
```

Comment:
Is the fix here the way this line is applied to a new wrapping container. Beforehand was it getting confused becase #container was setting a fixed width at the same time?

Reply:
The problem before was that at some point the layout had obviously gone from a single `div#container` to two nested `div.background > div#container`. There are two roles the styles were performing on the original `div#container`: styling the content of the container and styling the container itself. `body` is `display: flex`, and its child was `flex: 1 0 auto` meaning "grow me to 1 flex unit, never shrink me, and don't worry about an ideal size". With the introduction of `div.background`, we added an unstyled container between the `display: flex` and the `flex: 1 0 auto`, so `div.background` did the default behaviour for `display: flex` (no resizing), and `div#container` set an unused property `flex: 1 0 auto`. This change fixed that (while also giving more meaningful names to the selectors).